ext/parquet/types.go

```diff
@@ -0,0 +1,51 @@
+package parquet
+
+import (
+	"reflect"
+	"strings"
+	"sync"
+)
+
+type structFields map[string]int
+
+var fieldCache sync.Map // map[reflect.Type]structFields
```

Comment:
I see it's "inspired by json encode", but you really want it to be global and not per-decoder?

Reply:
nah, global is fine